app/controllers/katello/api/v2/upstream_subscriptions_controller.rb

```diff
@@ -0,0 +1,29 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class Api::V2::UpstreamSubscriptionsController < Api::V2::ApiController
+    before_action :find_organization
+
+    resource_description do
+      description "Red Hat subscriptions management platform."
+      api_version 'v2'
+    end
+
+    api :GET, "/organizations/:organization_id/upstream_subscriptions",
+      N_("List available subscriptions from Red Hat Subscription Management")
+    param :organization_id, :number, :desc => N_("Organization ID"), :required => true
+    param_group :search, Api::V2::ApiController
```

Comment:
Search isn't supported here, so we probably shouldn't include this

Reply:
looks like api controller doesn't split out paging params and search, so maybe you can split them out so that you can just reuse the paging stuff here (while still maintaining the search group the way it is)

Reply:
There are other differences. Here's the param group I'm thinking, which mimics the [Candlepin docs](http://www.candlepinproject.org/docs/gutterball/reportapi.html#pagination):


```ruby
    def_param_group :cp_search do
      param :page, :number, :desc => N_("Page number, starting at 1")
      param :per_page, :number, :desc => N_("Number of results per page to return.")
      param :order, String, :desc => N_("The order to sort the results in. ['asc', 'desc'] Defaults to 'desc'.")
      param :sort_by, String, :desc => N_("The field to sort the data by. Defaults to the created date.")
    end

```

Reply:
good from me after this fix.